pkg/kubelet/cm/node_container_manager_linux.go

```diff
@@ -222,6 +222,21 @@ func (cm *containerManagerImpl) GetNodeAllocatableReservation() v1.ResourceList
 	return result
 }
 
+// verifyReservedResourcesHaveCapacity validates reserved resources against the capacity of the resources collected from cAdvisor.
+func (cm *containerManagerImpl) verifyReservedResourcesHaveCapacity() error {
+	for k := range cm.NodeConfig.SystemReserved {
+		if _, ok := cm.capacity[k]; !ok {
+			return fmt.Errorf("cannot reserve %q resource", k)
+		}
+	}
+	for k := range cm.NodeConfig.KubeReserved {
+		if _, ok := cm.capacity[k]; !ok {
+			return fmt.Errorf("cannot reserve %q resource", k)
+		}
+	}
```

Comment:
Do we need to instead test whether the node has at least SystemReserved+KubeReserved of each resource?

Reply:
This is just to ensure that the node supports the resources being reserved. The amount is verified here: https://github.com/kubernetes/kubernetes/pull/83541/files#diff-1095a9691506e55223999753e484c972R513

Reply:
Can you please update the link? Because I can not find the verification that the number of reserved resources does not extend the node capacity.

Reply:
Hi @cynepco3hahue. This work is already handled by the container manager, by calling `validateNodeAllocatable` (just below the new `cm.validateNodeAllocatable` call). Hugepages are therefore "just like CPU and memory" in that sense. Reserving an amount giving a negative NodeAllocatable gives an error.

You can find the logic here: https://github.com/kubernetes/kubernetes/blob/442a69c3bdf6fe8e525b05887e57d89db1e2f3a5/pkg/kubelet/cm/node_container_manager_linux.go#L184-L250

Did that answer your question?

Reply:
Yep, absolutely, thanks